xcm-support/src/lib.rs

```diff
@@ -75,3 +78,75 @@ impl UnknownAsset for () {
 		Err(DispatchError::Other(NO_UNKNOWN_ASSET_IMPL))
 	}
 }
+
+/// Extracts the `AccountId32` from the passed `location` if the network
+/// matches.
+pub struct RelaychainAccountId32Aliases<Network, AccountId>(PhantomData<(Network, AccountId)>);
+impl<Network: Get<NetworkId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	xcm_executor::traits::Convert<MultiLocation, AccountId> for RelaychainAccountId32Aliases<Network, AccountId>
+{
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		let id = match location {
+			MultiLocation {
+				parents: 1,
+				interior: X1(AccountId32 {
+					id,
+					network: NetworkId::Any,
```

Comment:
This means any account on the parent network is going to be able to pass through this filter. The assumption here is that the parent network is always the relay chain, which doesn't hold true in all contexts. Instead, I propose adding another type parameter called `RelayChainLocation` to specifically define where exactly the relay chain is. It should be trait bounded with a `Contains`.

Reply:
This is also missing a match arm where the network ID here is equal to `Network::get()`, instead of being `Network::Any`.

Reply:
sounds reasonable